### PR TITLE
[BACK-814] Start of Device Status incorporation into the model

### PR DESCRIPTION
**Pull request by jamesraby**
This is the start of the device status implementation.

Tried to use interfaces as a better way of combining models.  Tests are not incorporated yet.

### CODE REVIEW

File: data/types/devicestatus/devicestatus.go [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+import (
+	"github.com/tidepool-org/platform/data"
+	"github.com/tidepool-org/platform/data/types"
+	"github.com/tidepool-org/platform/data/types/devicestatus/status"
+	"github.com/tidepool-org/platform/structure"
+)
+
+const (
+	Eid      = "eid"
```
Comment (pazaan) [inline data/types/devicestatus/devicestatus.go]: Should be "AID"

File: data/forecast.go [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+}
+
+func NewForecast() *Forecast {
+	return &Forecast{}
+}
+
+func (f *Forecast) Parse(parser structure.ObjectParser) {
+}
+
+func (f *Forecast) Validate(validator structure.Validator) {
```
Comment (pazaan) [inline data/forecast.go]: `startTime`, `timeScale`, `unit` and `values` are required. Also, `startTime` should parse out as a valid ISO 8601 timestamp

File: data/forecast.go [hunk 3]
```diff
@@ -0,0 +1,34 @@
+package data
```
Comment (pazaan) [inline data/forecast.go]: There should be tests for forecast.

File: data/types/devicestatus/devicestatus_test.go [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+var _ = Describe("DeviceStatus", func() {
+
+	Context("ParseDeviceStatus", func() {
+		// TODO
+	})
+
+	Context("NewDeviceStatus", func() {
+	})
+
+	Context("DeviceStatus", func() {
```
Comment (pazaan) [inline data/types/devicestatus/devicestatus_test.go]: Needs tests.

File: data/types/devicestatus/status/battery.go [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+func NewBattery() *Battery {
+	return &Battery{}
+}
+func (b *Battery) Parse(parser structure.ObjectParser) {
+	b.Unit = parser.String("unit")
+	b.Value = parser.Float64("value")
+}
+
+func (b *Battery) Validate(validator structure.Validator) {
+	validator.String("unit", b.Unit).Exists()
```
Comment (pazaan) [inline data/types/devicestatus/status/battery.go]: Value is also required.

File: data/types/devicestatus/status/reservoirremaining.go [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+	return &ReservoirRemaining{}
+}
+
+func (r *ReservoirRemaining) Parse(parser structure.ObjectParser) {
+	r.Unit = parser.String("unit")
+	r.Amount = parser.Float64("value")
+}
+
+func (r *ReservoirRemaining) Validate(validator structure.Validator) {
+}
```
Comment (pazaan) [inline data/types/devicestatus/status/reservoirremaining.go]: `unit` and `amount` should be required - that's my mistake. I only just thought that through, sorry :(

File: data/types/devicestatus/status/signalstrength.go [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+	return &SignalStrength{}
+}
+
+func (s *SignalStrength) Parse(parser structure.ObjectParser) {
+	s.Unit = parser.String("unit")
+	s.Value = parser.Float64("value")
+}
+
+func (s *SignalStrength) Validate(validator structure.Validator) {
+}
```
Comment (pazaan) [inline data/types/devicestatus/status/signalstrength.go]: `unit` and `value` should be required - that's my mistake. I only just thought that through, sorry :(

**Comment by jamesraby** [comment]
@pazaan - added more tests for device status objects existence.

File: data/types/devicestatus/status/signalstrength.go (hunk 7 above)
Comment (jamesraby) [reply data/types/devicestatus/status/signalstrength.go]: Added tests for unit and value

File: data/types/devicestatus/status/reservoirremaining.go (hunk 6 above)
Comment (jamesraby) [reply data/types/devicestatus/status/reservoirremaining.go]: Added tests for unit and amount

File: data/types/devicestatus/status/battery.go (hunk 5 above)
Comment (jamesraby) [reply data/types/devicestatus/status/battery.go]: Added tests for unit and value

File: data/types/devicestatus/devicestatus_test.go [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+
+var _ = Describe("DeviceStatus", func() {
+
+	Context("DeviceStatus", func() {
+		Context("Parse", func() {
+			// TODO
+		})
+
+		Context("Validate", func() {
+			DescribeTable("return the expected results when the input",
```
[the remark this answers is not in the document]
Comment (jamesraby) [reply data/types/devicestatus/devicestatus_test.go]: added tests

File: data/forecast.go (hunk 3 above)
Comment (jamesraby) [reply data/forecast.go]: added

**Pull request merged**